pkg/fanal/image/daemon/image.go

```diff
@@ -240,13 +240,20 @@ func emptyLayer(history dimage.HistoryResponseItem) bool {
 		strings.HasPrefix(createdBy, "VOLUME") ||
 		strings.HasPrefix(createdBy, "STOPSIGNAL") ||
 		strings.HasPrefix(createdBy, "SHELL") ||
-		strings.HasPrefix(createdBy, "ARG") ||
-		// buildkit image's layers with "WORKDIR /" command are empty
-		(createdBy == "WORKDIR /" && history.Comment == "buildkit.dockerfile.v0") {
+		strings.HasPrefix(createdBy, "ARG") {
 		return true
 	}
-	// commands here: 'ADD', COPY, RUN and WORKDIR != "/"
+	// commands here: 'ADD', COPY, RUN
 	// Also RUN command may not include 'RUN' prefix
 	// e.g. '/bin/sh -c mkdir test '
-	return false
+	var isEmptyWorkDir bool
+	// buildkit layers with "WORKDIR /" command are empty,
+	if history.Comment == "buildkit.dockerfile.v0" {
```

Comment:
I'm concerned about the case where the comment will be like `buildkit.dockerfile.v1`.
```suggestion
	if strings.HasPrefix(history.Comment, "buildkit.dockerfile") {
```